src/data/types.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * The search result object
+ * @typedef {Object} MediaResult
+ * @property {number} result_count
+ * @property {number} page_count
+ * @property {number} page_size
+ * @property {MediaResult[]} results
+ */
+
+/**
+ * @typedef {Object} APIResponse
+ * @property {Array} data
+ */
+export {}
```

Comment:
It looks like we're using `APIResponse` as a generic type here: https://github.com/WordPress/openverse-frontend/pull/173/files#diff-a6daedb006e158dcd94eb06a42f670f76b3c8ffe49b9dfdee53a53f6c492c0a9R51

As well in some other places.

We should expand the definition in that case to include a template parameter:

```suggestion
/**
 * @template T
 * @typedef {Object} APIResponse
 * @property {Array<T>} data
 */
export {}
```

I'm just guessing on the implementation of the generic parameter